src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -118,14 +163,66 @@ protected void processAccountingDocumentFromXml(String fileName, CreateAccountin
             removeDoneFileQuietly(fileName);
             createAndEmailReport(reportItem);
         }
-        if (reportItem.isNonBusinessRuleFailure()) {
+        if (reportItem.isDuplicateFile() && shouldExcludePreviouslyProcessedFiles()) {
+            logReport.getExcludedDuplicateFiles().add(fileName);
+            logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
+        } else if (reportItem.isNonBusinessRuleFailure()) {
             logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
         } else {
             logReport.getFilesSuccessfullyProcessed().add(fileName);
         }
         LOG.info("processAccountingDocumentFromXml: Value of reportItem.isNonBusinessRuleFailure just prior to method return =" + reportItem.isNonBusinessRuleFailure());
     }
-    
+
+    protected CreateAccountingDocumentFileEntry findExistingFileEntry(String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        Map<String, String> criteria = Map.of(KFSPropertyConstants.FILE_NAME, cleanedFileName);
+        Collection<CreateAccountingDocumentFileEntry> fileEntries = businessObjectService.findMatching(
+                CreateAccountingDocumentFileEntry.class, criteria);
+        return CollectionUtils.isNotEmpty(fileEntries) ? fileEntries.iterator().next() : null;
+    }
+
+    protected String getCleanedFileName(String fileName) {
+        String cleanedFileName = fileName;
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.BACKSLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.BACKSLASH);
+        }
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.SLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.SLASH);
+        }
+        cleanedFileName = StringUtils.lowerCase(cleanedFileName, Locale.US);
+        return cleanedFileName;
+    }
+
+    private boolean shouldExcludePreviouslyProcessedFiles() {
+        return parameterService.getParameterValueAsBoolean(KFSConstants.CoreModuleNamespaces.FINANCIAL, 
+                CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCOUNTING_DOCUMENT_SERVICE_COMPONENT_NAME, 
+                CuFPParameterConstants.CreateAccountingDocumentService.DUPLICATE_FILE_CHECK_IND);
+    }
+
+    private AccountingXmlDocumentListWrapper unmarshalAccountingDocumentFile(String fileName) {
+        byte[] fileData = LoadFileUtils.safelyLoadFileBytes(fileName);
+        return (AccountingXmlDocumentListWrapper) batchInputFileService.parse(
+                accountingDocumentBatchInputFileType, fileData);
+    }
+
+    private void createFileEntry(AccountingXmlDocumentListWrapper accountingXmlDocuments, String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        Long nextFileId = sequenceAccessorService.getNextAvailableSequenceNumber(

```

Comment:
Is this needed?  I thought BusinessObjectService / OJB would set this if not set.

Reply:
I'll see if I can set this up through OJB metadata instead of the SequenceAccessorService.

Reply:
I don't think you need to set the fileId value, won't OJB set it atomically from the sequence?

Reply:
Yes, I meant to say that I'm changing this to use just the OJB XML for this setup, and that I'll remove the manual ID-populating code (at least on the non-unit-test side). Sorry for the confusion.